plugin/keyring_vault/vault_curl.cc

```diff
@@ -0,0 +1,177 @@
+#include <algorithm>
+#include "vault_curl.h"
+#include "base64.h"
+
+namespace keyring
+{
+
+static size_t write_response_memory(void *contents, size_t size, size_t nmemb, void *userp)
```

Comment:
Please, add some logic here which would limit max HTTP response data size to a reasonable value, say 4K so that malicious Vault servers won't cause OOMs.